File: benchmark/src/main/java/de/hhu/bsinfo/observatory/benchmark/BenchmarkPhase.java
package de.hhu.bsinfo.observatory.benchmark;

import de.hhu.bsinfo.observatory.benchmark.result.Status;

abstract class BenchmarkPhase {

    private final Benchmark benchmark;

    BenchmarkPhase(Benchmark benchmark) {
        this.benchmark = benchmark;
    }

    Benchmark getBenchmark() {
        return benchmark;
    }

    abstract Status execute();
}

File: cpp/src/observatory/BenchmarkPhase.cpp
#include <observatory/result/Status.h>
#include <memory>
#include <utility>
#include "BenchmarkPhase.h"

namespace Observatory {

BenchmarkPhase::BenchmarkPhase(Benchmark &benchmark) :
        benchmark(benchmark) {}

Benchmark& BenchmarkPhase::getBenchmark() const {
    return benchmark;
}

}
